### ISSUE TITLE
Visibility in Draw components doesn't work properly, I suppose

### ISSUE THREAD

**Issue by jayong93** [bevyengine/bevy#187]
First of all, thanks for the great game engine!

I have used it to make my prototype game. In the game, I wanted to hide some object(entity) in a particular condition, so I changed the `is_visible` value in the Draw component of the entity.

But when I set the `is_visible` value to `false`, the frame buffer suddenly wasn't refreshed at all. The window kept showing me the last frame just before I set the value to false. And I set the value to true again, then whole world goes normally.

Is this intended or just a bug? If it is a bug, would you fix it or let me know where to start fixing?

**Comment by karroffel** [comment]
(I assigned the `bug` label because it *sounds* like a bug to me, but I am not fully sure myself :sweat_smile: )

**Comment by cart** [comment]
Yeah this certainly sounds like a bug. Individual entity visibility shouldn't determine whether or not we clear the framebuffer. A minimal example that reproduces the error would be helpful here.

**Comment by jayong93** [comment]
@cart 
I added some code to the bevy's breakout example to reproduce the bug:
https://gist.github.com/jayong93/6d2082442c8f7cfe5a4c89b532f98940

What I did is adding this system to `breakout.rs` to hide a ball on mouse click:
```rust
fn speed_circle_place_system(
    mut input_reader: ResMut<EventReader<MouseButtonInput>>,
    button_input: Res<Events<MouseButtonInput>>,
    mut query: Query<(&Ball, &mut Draw)>,
) {
    match input_reader.latest(&button_input) {
        Some(MouseButtonInput {
            button: MouseButton::Left,
            state: ElementState::Pressed,
        }) => {
            for (_, mut draw) in &mut query.iter() {
                draw.is_visible = !draw.is_visible;
            }
        }
        _ => {}
    }
}
```
If you click left mouse button, the ball would disappear. But also the paddle would stop moving even if you move it by pressing left/right arrow key.

I checked this bug on Windows 10 and I'm not sure that it is reproducible on other OSs.

**Comment by Bobox214** [comment]
Issue reproduced on Linux with @jayong93 code, with rustc 1.47.0-nightly (6c8927b0c 2020-07-26) and bevy at 6228e8a4bbdb966659f7dae1e87cf55e8318bde3

**Comment by cart** [comment]
Resolved by #230
